test/e2e_node/swap_test.go

```diff
@@ -38,49 +45,231 @@ import (
+				ginkgo.Entry("QOS Guaranteed", v1.PodQOSGuaranteed, false),
+			)
+		})
+
+		// These tests assume the following, and will fail otherwise:
+		// - The node is provisioned with swap
+		// - The node is configured with cgroup v2
+		// - The swap feature gate is enabled
+		// - The node has no more than 15GB of memory
+		f.Context("with swap stress", func() {
+			var (
+				nodeName           string
+				nodeTotalMemory    *resource.Quantity
+				nodeUsedMemory     *resource.Quantity
+				swapCapacity       *resource.Quantity
+				stressMemAllocSize *resource.Quantity
+				podClient          *e2epod.PodClient
+			)
+
+			ginkgo.BeforeEach(func() {
+				podClient = e2epod.NewPodClient(f)
+
+				sleepingPod := getSleepingPod(f.Namespace.Name)
+				sleepingPod = runPodAndWaitUntilScheduled(f, sleepingPod)
+				gomega.Expect(isPodCgroupV2(f, sleepingPod)).To(gomega.BeTrueBecause("node uses cgroup v1"))
+
+				nodeName = sleepingPod.Spec.NodeName
+				gomega.Expect(nodeName).ToNot(gomega.BeEmpty(), "node name is empty")
+
+				nodeTotalMemory, nodeUsedMemory = getMemoryCapacity(f, nodeName)
+				gomega.Expect(nodeTotalMemory.IsZero()).To(gomega.BeFalseBecause("node memory capacity is zero"))
+				gomega.Expect(nodeUsedMemory.IsZero()).To(gomega.BeFalseBecause("node used memory is zero"))
+
+				swapCapacity = getSwapCapacity(f, sleepingPod)
+				gomega.Expect(swapCapacity.IsZero()).To(gomega.BeFalseBecause("node swap capacity is zero"))
+
+				err := podClient.Delete(context.Background(), sleepingPod.Name, metav1.DeleteOptions{})
+				framework.ExpectNoError(err)
+
+				stressMemAllocSize = multiplyQuantity(divideQuantity(nodeTotalMemory, 1000), 4)
+
+				ginkgo.By(fmt.Sprintf("Setting node values. nodeName: %s, nodeTotalMemory: %s, nodeUsedMemory: %s, swapCapacity: %s, stressMemAllocSize: %s",
+					nodeName, nodeTotalMemory.String(), nodeUsedMemory.String(), swapCapacity.String(), stressMemAllocSize.String()))
+			})
+
+			ginkgo.Context("LimitedSwap", func() {
+				tempSetCurrentKubeletConfig(f, enableLimitedSwap)
+
+				getRequestBySwapLimit := func(swapPercentage int64) *resource.Quantity {
+					gomega.ExpectWithOffset(1, swapPercentage).To(gomega.And(
+						gomega.BeNumerically(">=", 1),
+						gomega.BeNumerically("<=", 100),
+					), "percentage has to be between 1 and 100")
+
+					// <swap-limit> = <memory-request>*(<swap-capacity>/<node-capacity>).
+					// if x is the percentage, and <memory-request> == (x/100)*<node-capacity>, then:
+					// <swap-limit> = (x/100)*<node-capacity>*(<swap-capacity>/<node-capacity>) = (x/100)*<swap-capacity>.
+					return multiplyQuantity(divideQuantity(nodeTotalMemory, 100), swapPercentage)
+				}
+
+				getStressPod := func(stressSize *resource.Quantity) *v1.Pod {
+					pod := getStressPod(f, stressSize, stressMemAllocSize)
+					pod.Spec.NodeName = nodeName
+
+					return pod
+				}
+
+				ginkgo.It("should be able over-commit the node memory", func() {
```

Comment:
minor: over-commit sounds like we can request more than node allows. I'd call it should be able to use more than the node memory capacity

Reply:
Done!